### ISSUE TITLE
Q1 2022 priorities

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#2492]
In the first quarter of 2022, our priorities are:

* Implementing the [C++20 Defect Reports](https://github.com/microsoft/STL/projects/9?fullscreen=true)
  + And then backporting them to VS 2019 16.11.x.
* Reviewing the PR backlog
* Implementing the [C++23 features](https://github.com/microsoft/STL/projects/8?fullscreen=true) that have been accepted so far
* Writing Contribution Guidelines

We could use some help with #2475 if anyone has free time! :smile_cat:

If you have questions about our plans for this quarter, feel free to ask them here. (Note that compiler, IDE, etc. priorities are off-topic.)

**Comment by jonwil** [comment]
Has the plan to open source vcruntime/vcstartup been dropped or is it just delayed (e.g. legal reviews, code cleanup, whatever
)

**Comment by StephanTLavavej** [comment]
Only delayed - we need to create some benchmark infrastructure before the compiler back-end team will be comfortable accepting changes to its perf-critical functions.

**Comment by jovibor** [comment]
Can you please confirm that 17.2 release will mark the `C++20 ABI` lockdown (due to improbable urgency the #2518 was closed),
and that we won't see any `format` and `ranges` improvements until `vNext`? 😭😭😎

**Comment by StephanTLavavej** [comment]
Yes, 17.2 will contain the C++20 ABI lockdown - as decided by our bosses and boss-like entities.

We can continue to improve `<format>` and `<ranges>` in ways that preserve ABI.

**Comment by StephanTLavavej** [comment]
@AlexGuteniev Thanks for asking! @barcharcraz says that #2119 is definitely not ABI breaking because it doesn't change representations of data structures, and that #2437 can be made to be non-ABI-breaking by renaming internal data structures (e.g. from `_Ugly` to `_Ugly2`) so I think we should be okay there.

**Comment by jrmoore** [comment]
What is missing from `<ranges>`? From the blog post titled [MSVC C++20 and the /std:c++20 Switch](https://devblogs.microsoft.com/cppblog/msvc-cpp20-and-the-std-cpp20-switch/) it seems [P0896R4](https://wg21.link/P0896R4) was implemented already for VS 2019 v16.10 & v16.11, but `__cpp_lib_ranges` seems not to be defined.

Although it also states `/std:c++20` to be ABI stable and from what I'm reading that won't happen until v17.2 😅.

**Comment by CaseyCarter** [comment]
> What is missing from `<ranges>`? From the blog post titled [MSVC C++20 and the /std:c++20 Switch](https://devblogs.microsoft.com/cppblog/msvc-cpp20-and-the-std-cpp20-switch/) it seems [P0896R4](https://wg21.link/P0896R4) was implemented already for VS 2019 v16.10 & v16.11, but `__cpp_lib_ranges` seems not to be defined.
> 
> Although it also states `/std:c++20` to be ABI stable and from what I'm reading that won't happen until v17.2 😅.

You can look at #1814 for full gory detail, but the short version is "We couldn't put `<ranges>`/`<format>` under `/std:c++20` until WG21 stopped breaking ABI with Defect Report fixes." If you compile with `/std:c++latest` - the flag that explicitly asks for unstable features from in-progress implementation of the not-yet-complete next version of the C++ standard - you will see `__cpp_lib_ranges` is defined. But be aware that there are significant ABI-breaking changes going on until 17.2 p2 when we finally pull `<ranges>`/`<format>` into `/std:c++20`.

**Comment by StephanTLavavej** [comment]
Q2 2022 priorities are now pinned: #2628
